service-base/src/main/java/org/eclipse/hono/service/registration/BaseRegistrationService.java

```diff
@@ -124,28 +133,34 @@ protected String getEventBusAddress() {
         final String gatewayId = request.getGatewayId();
         final SpanContext spanContext = request.getSpanContext();
 
+        final Span span = newChildSpan(SPAN_NAME_ASSERT_DEVICE_REGISTRATION, spanContext, tenantId, deviceId, gatewayId);
         if (tenantId == null || deviceId == null) {
+            TracingHelper.logError(span, "missing tenant and/or device");
+            span.finish();
             return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST));
-        } else if (gatewayId == null) {
-            log.debug("asserting registration of device [{}] with tenant [{}]", deviceId, tenantId);
-            final Future<RegistrationResult> result = Future.future();
-            assertRegistration(tenantId, deviceId, spanContext, result.completer());
-            return result.map(res -> {
-                return request.getResponse(res.getStatus())
-                        .setDeviceId(deviceId)
-                        .setJsonPayload(res.getPayload())
-                        .setCacheDirective(res.getCacheDirective());
-            });
         } else {
-            log.debug("asserting registration of device [{}] with tenant [{}] for gateway [{}]",
-                    deviceId, tenantId, gatewayId);
             final Future<RegistrationResult> result = Future.future();
-            assertRegistration(tenantId, deviceId, gatewayId, spanContext, result.completer());
+            if (gatewayId == null) {
+                log.debug("asserting registration of device [{}] with tenant [{}]", deviceId, tenantId);
+                assertRegistration(tenantId, deviceId, span, result.completer());
+            } else {
+                log.debug("asserting registration of device [{}] with tenant [{}] for gateway [{}]",
+                        deviceId, tenantId, gatewayId);
+                assertRegistration(tenantId, deviceId, gatewayId, span, result.completer());
+            }
             return result.map(res -> {
+                if (res.isError()) {
+                    TracingHelper.logError(span, "status " + res.getStatus());
```

Comment:
we usually use `io.opentracing.tag.Tags.HTTP_STATUS` to record the resulting status code of invoking a service operation. Marking the Span as erroneous is still necessary, though, but could be done with simply invoking
`io.opentracing.tag.Tags.ERROR.set(span, true)`

Reply:
Ok, I've added a tag for `HTTP_STATUS`.